### PR TITLE
Swagger Documentation for SelfLoansApiResource.java

### PR DESCRIPTION
**Pull request by luckyman20**
I haven't annotated the last POST endpoint as it is not in the existing documentation

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java [hunk 1]
```diff
@@ -143,9 +145,11 @@ public String retrieveLoanCharge(@PathParam("loanId") final Long loanId,
     @Path("template")
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String template(@QueryParam("clientId") final Long clientId, 
-    		@QueryParam("productId") final Long productId,
-    		@QueryParam("templateType") final String templateType,
+	@ApiOperation(value = "Retrieve Loan Details Template", httpMethod = "GET", notes = "Retrieves Loan Details Template\n\n" + "This is a convenience resource. It can be useful when building maintenance user interface screens for client applications. The template data returned consists of any or all of:\n" + "\n" + "Field Defaults\n" + "Allowed Value Lists\n\n" + "Example Requests:\n" + "\n" + "self/loans/template?templateType=individual&clientId=1\n" + "\n" + "\n" + "self/loans/template?templateType=individual&clientId=1&productId=1")
+	@ApiResponses({@ApiResponse(code = 200, message = "OK", response = SelfLoansApiResourceSwagger.GetSelfLoansTemplateResponse.class)})
+    public String template(@QueryParam("clientId") @ApiParam(value = "clientId") final Long clientId,
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java]: indentation of @ApiOperation and @ApiResponses are not correct

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java [hunk 2]
```diff
@@ -171,8 +175,11 @@ public String template(@QueryParam("clientId") final Long clientId,
     @POST
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java:175]: Incomplete

**Review by Sanyam96 — CHANGES_REQUESTED** [review]
some indentation of Annotations is not correct!!

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java (hunk 2 above)
Comment (luckyman20) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java:175]: What is incomplete here?

Comment (Sanyam96) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java:175]: all post methods are not mentioned!
refer to the current documentation

Comment (luckyman20) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java:175]: Yes, Thats what I have mentioned in the description, only documentation of one POST endpoint is there so I have added for only one

Comment (Sanyam96) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java:175]: you have to mention like previous examples separated by '|' character 
and also the indentation is not correct

**Review by Sanyam96 — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java [hunk 3]
```diff
@@ -171,8 +175,11 @@ public String template(@QueryParam("clientId") final Long clientId,
...
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String calculateLoanScheduleOrSubmitLoanApplication(@QueryParam("command") final String commandParam,
-            @Context final UriInfo uriInfo, final String apiRequestBodyAsJson) {
+    @ApiOperation(value = "Applicant Withdraws from Loan Application | Calculate Loan Repayment Schedule", httpMethod = "POST", notes = "Applicant Withdraws from Loan Application\n\n" + "Mandatory Fields: withdrawnOnDate\n\n" + "Calculate Loan Repayment Schedule:\n\n" + "Calculates Loan Repayment Schedule\n\n" + "Mandatory Fields: productId, principal, loanTermFrequency, loanTermFrequencyType, numberOfRepayments, repaymentEvery, repaymentFrequencyType, interestRatePerPeriod, amortizationType, interestType, interestCalculationPeriodType, expectedDisbursementDate, transactionProcessingStrategyId")
```
Comment (Sanyam96) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/self/loanaccount/api/SelfLoansApiResource.java]: incorrect value: "Calculate Loan Repayment Schedule | Submit a new Loan Application" and notes

**Review by Sanyam96 — CHANGES_REQUESTED** [review]

**Comment by luckyman20** [comment]
@Sanyam96 I have updated the PR

**Review by Sanyam96 — CHANGES_REQUESTED** [review]
annotations missing for POST endpoint line 211
https://github.com/Sanyam96/incubator-fineract/pull/17/commits/9d4c18e7fac1567b02be44d8fbe4c860a2ae9e84#diff-c902b46393b4c91a0aa9789ce725f9e0R211

Please complete this annotation
Rest all looks good

And make a new commit of it
don't merge with the previous commit

**Comment by Sanyam96** [comment]
Make a new commit
don't merge 2 commits

**Comment by luckyman20** [comment]
@Sanyam96 Updated the PR

**Review by Sanyam96 — APPROVED** [review]

**Pull request merged**